test/validators/register.ts

```diff
@@ -1,53 +1,133 @@
+            helpers.DataGenerator.publicKey(0),
+            [1, 2],
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.be.revertedWith('OessDataStructureInvalid');
+
+        // 6 Operators
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 4, 5, 6],
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.be.revertedWith('OessDataStructureInvalid');
+    });
+
+    it('Invalid public key length', async () => {
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.shares(0),
+            [1, 2, 3, 4],
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.be.revertedWith('InvalidPublicKeyLength');
+    });
+
+    it('Not enough amount', async () => {
+        await expect(ssvNetworkContract.connect(helpers.DB.owners[1]).registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 4],
+            helpers.DataGenerator.shares(0),
+            '1'
+        )).to.be.revertedWith('account liquidatable');
+    });
+
+    // GAS AMOUNT IS ABOVE EXPECTED
+    it('Register with 7 operators', async () => {
+        await helpers.registerValidators(0, 1, '10000', helpers.DataGenerator.pod.new(7), [GasGroup.REGISTER_VALIDATOR_EXISTED_POD]);
+    });
+
+    // THIS NEEDS VALIDITY (I ADDED IN SOLIDITY CODE)
+    it('Non existent operator', async () => {
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 25],
+            helpers.DataGenerator.shares(0),
+            '4000'
+        )).to.be.revertedWith('OperatorDoesntExist');
+    });
+
+    // THIS NEEDS VALIDITY (I ADDED IN SOLIDITY CODE)
+    it('Register with existing validator', async () => {
+        await expect(ssvNetworkContract.connect(helpers.DB.owners[1]).registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 4],
+            helpers.DataGenerator.shares(0),
+            '4000'
+        )).to.be.revertedWith('ValidatorAlreadyExists');
+    });
+
+    // THIS NEEDS SOME PROPER ERROR MESSAGE (COULDNT GET ERROR MESSAGE TO WORK HERE WHEN I ADDED LOGIC IN SOLIDITY)
+    it('Not enough deposited', async () => {
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 4],
+            helpers.DataGenerator.shares(0),
+            '100005'
+        )).to.be.revertedWith('NotEnoughDeposited');
+    });
+
+    // MAYBE WE WILL ADD SOME VALIDITY HERE?
+    // it('Invalid share', async () => {
+    //     await expect(ssvNetworkContract.registerValidator(
+    //         helpers.DataGenerator.publicKey(0),
+    //         [1, 2, 3, 4],
+    //         helpers.DataGenerator.publicKey(0),
+    //         '10000'
+    //     )).to.be.revertedWith('InvalidShareLength');
+    // });
```

Comment:
@lior-blox Maybe in the future to add some sort of validity here?  If not lets delete it in the PR